### PR TITLE
better naming for Secure Boot options (issue#866)

### PR DESCRIPTION
**Pull request by jsmeix**
renamed recovery system boot menue entries
from 'Secure Boot' to 'UEFI boot' and
from 'no Secure Boot' to 'no UEFI - usually BIOS boot'
see https://github.com/rear/rear/issues/866

### CODE REVIEW

**Comment by gdha** [comment]
@jsmeix @gozora let us wait on the tests performed by @gozora before deciding to commit this one

**Comment by jsmeix** [comment]
I changed the names to no longer so nice looking
but (hopefully) technically more correct ones.

Now the following menue entries exist
(grep for 'Relax and Recover (' in the sources):
<pre>
Relax and Recover (using elilo)
Relax and Recover (using GRUB 2 'linux' and 'initrd')
Relax and Recover (using GRUB 2 'linuxefi' and 'initrdefi')
Relax and Recover (using GRUB Legacy)
Relax and Recover (using grub 0.97)
Relax and Recover (using grub 2.0 'linux' and 'initrd')
</pre>
Ugly but hopefully technically correct.

**Comment by schlomo** [comment]
So do we actually support UEFI Secure Boot?

**Comment by gdha** [comment]
@schlomo @jsmeix Secure boot is the entry using `linuxefi`, but will some of these entries will not be valid I guess when there is no executable behind to kick off, or do I misunderstood it?

**Comment by jsmeix** [comment]
For me the info from @abbbi in
https://github.com/rear/rear/issues/866#issuecomment-224563239
was most interesting because it seems to prove that
sometimes only the GRUB 2 'linux'/'initrd' method works
and sometimes only the GRUB 2 'linuxefi'/'initrdefi' method works.

Unfortunately @abbi did not mention
if he uses UEFI with or without Secure Boot.

Because I have currently no idea under what conditions
which boot method works, I removed words like "UEFI"
and "Secure Boot" to avoid false expectations and
used only the technical info what bootloader method
is used.

In particular I used different texts for each of those
bootloader methods in rear (according to the comments
that I found in the sources) so that we could see in
user issue reports what exact method was used.

**Comment by gdha** [comment]
@jsmeix has the issue #866 cleared out regarding this pull request? I'm a bit lost in the case I'm afraid

**Comment by jsmeix** [comment]
@gdha 
no news in https://github.com/rear/rear/issues/866
and currently I don't have the time to dig into it on my own
to get a better basic understnding of Secure Boot.

Accordingly I set the milestone to any unspecified
future rear version.

**Comment by gdha** [comment]
@jsmeix is this pull request still valid?

**Comment by jsmeix** [comment]
With https://github.com/rear/rear/pull/955 merged
this pull request could be obsolete.

But I would like to wait what @abbbi reports
in https://github.com/rear/rear/issues/866
if with https://github.com/rear/rear/pull/955 merged
really both both GRUB menue entries
"Relax and Recover (no Secure Boot)" and
"Relax and Recover (Secure Boot)" are working.

As far as I understand what @abbbi wrote in
https://github.com/rear/rear/issues/866#issuecomment-224563239
<pre>
SLES12: (No Secureboot) fails with "can't find command linux"
</pre>
and according to what I wrote in
https://github.com/rear/rear/issues/866#issuecomment-224571520
the current naming
"Relax and Recover (no Secure Boot)" and
"Relax and Recover (Secure Boot)"
seems to be not really correct as far as I understand it
because
"Relax and Recover (no Secure Boot)" does
<pre>
linux /isolinux/kernel ...
initrd /isolinux/initrd.cgz
</pre>
and
"Relax and Recover (Secure Boot)" does
<pre>
linuxefi /isolinux/kernel ...
initrdefi /isolinux/initrd.cgz
</pre>
i.e. different GRUB commands are called
and those GRUB commands linux/initrd versus
linuxefi/initrdefi differ regarding whether or not (U)EFI
is used and not whether or not Secure Boot is used
so that - as far as I understand it - my proposal to
rename those GRUB menue entries into
"Relax and Recover (using GRUB 2 'linux' and 'initrd')" and
"Relax and Recover (using GRUB 2 'linuxefi' and 'initrdefi')"
is still valid from my point of view.

Bottom line:
As long as it is not clear that the GRUB commands
linux/initrd match exactly "no Secure Boot" and
linuxefi/initrdefi match exactly "Secure Boot"
the current GRUB menue entries names
"Relax and Recover (no Secure Boot)" and
"Relax and Recover (Secure Boot)"
should be replaced by correct names.

**Comment by jsmeix** [comment]
According to what @gozora explained in
https://github.com/rear/rear/issues/866#issuecomment-238493050
it now seems that the current GRUB menue entries names
"Relax and Recover (no Secure Boot)" and
"Relax and Recover (Secure Boot)" are the right ones
so that this pull request is obsolete.
